### ISSUE TITLE
Base types depend on sub package types

### ISSUE THREAD

**Issue by boxleytw** [klogging/klogging#10]
Having types in a higher-level package depend on types in a subpackage is considered a design smell when code segregates by package structure, as is typically the case in modern library design.  Not sure if this is a guideline you want to follow.

The case I spot off the bat is:
```
import ktlogging.events.Level

interface BaseLogger {
// PREVIOUS CODE ELIDED
    fun isTraceEnabled() = minLevel() <= Level.TRACE
// LATER CODE ELIDED
```

```
enum class Level {
    NONE, TRACE, DEBUG, INFO, WARN, ERROR, FATAL
}
```

In this case, `BaseLogger` in `ktlogging` depends on `Level` in `ktlogging.events`.

Of course this guideline is /not/ a rule, and can be hard to follow sometimes without making a muck of things.

The "fix" in this case is trivial: move `Logging` to the `ktlogging` package.  I'm happy to make this change if it aligns with your design ideas for the project, in which case, I'll also run IDEA analysis to find other cases like this.

**Comment by mjstrasser** [comment]
Thanks @boxleytw again!

I am implementing explicit API mode right now, so please hold off from making changes while I do that.
